### ISSUE TITLE
running on relative ansible root doesn't honor ansible.cfg

### ISSUE THREAD

**Issue by nrobert13** [ansible/ansible-lint#2611]
##### Summary

Running ansible-lint on relative folder doesn't detect the roles.
I have a git repo which contains a folder ansible, where all the ansible related files are placed

repo
 |- ansible
     |- ansible.cfg
     |- playbooks/ssh-keys.yml
     |- roles/ssh-keys/...

if I run `ansible-lint ansible` from the `repo` folder, I get a totally different output than running `ansible-lint` in the `ansible` folder.

```
> ansible-lint ansible
WARNING  Listing 1 violation(s) that are fatal
syntax-check[specific]: the role 'ssh-keys' was not found in /Users/rnemeti/test/ansible/playbooks/roles:/Users/rnemeti/.cache/ansible-compat/9f86d0/roles:/Users/rnemeti/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:/Users/rnemeti/test/ansible/playbooks
ansible/playbooks/ssh-keys.yml:8:7

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - syntax-check[specific]  # Ansible syntax check failed.

                  Rule Violation Summary
 count tag                    profile rule associated tags
     1 syntax-check[specific] min     core

Failed after : 1 failure(s), 0 warning(s) on 4 files.

```

if I run inside `ansible` folder, I get the expected output:

```
> ansible-lint
[WARNING]: Unable to find 'sshkeys/syseleven.pub' in expected paths (use -vvvvv to see paths)
[WARNING]: Unable to find 'sshkeys/vault-ci-stage.pub' in expected paths (use -vvvvv to see paths)
[WARNING]: Unable to find 'sshkeys/vault-ci-prod.pub' in expected paths (use -vvvvv to see paths)
WARNING  Listing 15 violation(s) that are fatal
name[play]: All plays should be named. (warning)
playbooks/ssh-keys.yml:1

schema[playbook]: '{{ any_errors_fatal | default(true) }}' is not of type 'boolean' (warning)
playbooks/ssh-keys.yml:1  Returned errors will not include exact line numbers, but they will mention
the schema name being used as a tag, like ``schema[playbook]``,
``schema[tasks]``.

This rule is not skippable and stops further processing of the file.

Schema bugs should be reported towards [schemas](https://github.com/ansible/schemas) project instead of ansible-lint.

If incorrect schema was picked, you might want to either:

* move the file to standard location, so its file is detected correctly.
* use ``kinds:`` option in linter config to help it pick correct file type.


yaml[truthy]: Truthy value should be one of [false, true]
playbooks/ssh-keys.yml:3

yaml[truthy]: Truthy value should be one of [false, true]
playbooks/ssh-keys.yml:5

role-name: Role name ssh-keys does not match ``^[a-z][a-z0-9_]*$`` pattern. (warning)
roles/ssh-keys:1

fqcn[action-core]: Use FQCN for builtin module actions (set_fact).
roles/ssh-keys/tasks/main.yml:2 Use `ansible.builtin.set_fact` or `ansible.legacy.set_fact` instead.

jinja[spacing]: Jinja2 spacing could be improved: {{ lookup('file',  'sshkeys/syseleven.pub' ) }} -> {{ lookup('file', 'sshkeys/syseleven.pub') }} (warning)
roles/ssh-keys/tasks/main.yml:2 Jinja2 template rewrite recommendation: `{{ lookup('file', 'sshkeys/syseleven.pub') }}`.

fqcn[action-core]: Use FQCN for builtin module actions (set_fact).
roles/ssh-keys/tasks/main.yml:6 Use `ansible.builtin.set_fact` or `ansible.legacy.set_fact` instead.

jinja[spacing]: Jinja2 spacing could be improved: {{ lookup('file',  'sshkeys/vault-ci-stage.pub' ) }} -> {{ lookup('file', 'sshkeys/vault-ci-stage.pub') }} (warning)
roles/ssh-keys/tasks/main.yml:6 Jinja2 template rewrite recommendation: `{{ lookup('file', 'sshkeys/vault-ci-stage.pub') }}`.

fqcn[action-core]: Use FQCN for builtin module actions (set_fact).
roles/ssh-keys/tasks/main.yml:11 Use `ansible.builtin.set_fact` or `ansible.legacy.set_fact` instead.

jinja[spacing]: Jinja2 spacing could be improved: {{ lookup('file',  'sshkeys/vault-ci-prod.pub' ) }} -> {{ lookup('file', 'sshkeys/vault-ci-prod.pub') }} (warning)
roles/ssh-keys/tasks/main.yml:11 Jinja2 template rewrite recommendation: `{{ lookup('file', 'sshkeys/vault-ci-prod.pub') }}`.

fqcn[action-core]: Use FQCN for builtin module actions (set_fact).
roles/ssh-keys/tasks/main.yml:16 Use `ansible.builtin.set_fact` or `ansible.legacy.set_fact` instead.

fqcn[action]: Use FQCN for module actions, such `<namespace>.<collection>.authorized_key`. (warning)
roles/ssh-keys/tasks/main.yml:23 Action `authorized_key` is not FQCN.

yaml[truthy]: Truthy value should be one of [false, true]
roles/ssh-keys/tasks/main.yml:27

fqcn[action-core]: Use FQCN for builtin module actions (file).
roles/ssh-keys/tasks/main.yml:29 Use `ansible.builtin.file` or `ansible.legacy.file` instead.

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - experimental  # all rules tagged as experimental
  - fqcn[action-core]  # Use FQCN for builtin actions.
  - yaml[truthy]  # Violations reported by yamllint.

                       Rule Violation Summary
 count tag               profile    rule associated tags
     3 jinja[spacing]    basic      formatting (warning)
     1 role-name         basic      deprecations, metadata (warning)
     1 schema[playbook]  basic      core, experimental (warning)
     1 name[play]        basic      idiom (warning)
     3 yaml[truthy]      basic      formatting, yaml
     5 fqcn[action-core] production formatting
     1 fqcn[action]      production formatting (warning)

Failed after min profile: 8 failure(s), 7 warning(s) on 4 files.

```

I also tried to create `ansible.cfg` in the `repo` folder like this
repo
  |- ansible.cfg
```
> cat ansible.cfg
roles_path = ansible/roles/
```
##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
> ansible --version
ansible [core 2.12.10]
  config file = /Users/rnemeti/test/ansible.cfg
  configured module search path = ['/Users/rnemeti/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/rnemeti/work/kubermatic/lib/python3.10/site-packages/ansible
  ansible collection location = /Users/rnemeti/.ansible/collections:/usr/share/ansible/collections
  executable location = /Users/rnemeti/work/kubermatic/bin/ansible
  python version = 3.10.6 (main, Aug 30 2022, 04:58:14) [Clang 13.1.6 (clang-1316.0.21.2.5)]
  jinja version = 3.1.2
  libyaml = False

> ansible-lint --version
ansible-lint 6.8.1 using ansible 2.12.10

```

- ansible installation method:  pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

```console (paste below)
> find . -type f | xargs -I% bash -c "echo %; cat %"
./ansible/roles/ssh-keys/tasks/main.yml
---
- name: Load ssh keys for the team
  set_fact:
    ssh_keys: "{{ lookup('file',  'sshkeys/keys.pub' ) }}"

- name: Load ssh key for CI (vault/stage)
  set_fact:
    vault_keys: "{{ lookup('file',  'sshkeys/vault-ci-stage.pub' ) }}"
  when: "'vault' in group_names and 'stage' in instance_name"

- name: Load ssh key for CI (vault/prod)
  set_fact:
    vault_keys: "{{ lookup('file',  'sshkeys/vault-ci-prod.pub' ) }}"
  when: "'vault' in group_names and 'seed' in instance_name"

- name: Add vault ssh key to the team keys
  set_fact:
    ssh_keys: |
      {{ ssh_keys }}
      {{ vault_keys }}
  when: vault_keys is defined

- name: Ensure authorized ssh keys
  authorized_key:
    user: ubuntu
    key: "{{ ssh_keys }}"
    exclusive: yes

- name: Remove Root authorized_keys
  file:
    path: /root/.ssh/authorized_keys
    state: absent
./ansible/playbooks/ssh-keys.yml
- hosts: all
  user: ubuntu
  become: yes
  become_method: sudo
  gather_facts: no
  any_errors_fatal: "{{ any_errors_fatal | default(true) }}"
  roles:
    - ssh-keys
./ansible/ansible.cfg
[defaults]
roles_path = roles/

```

##### Desired Behavior

get same output running `ansible-lint ansible` in the `repo` folder and running `ansible-lint` in the `repo/ansible` folder.

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

getting a totally different output.

Please give some details of what is actually happening.
Include a [minimum complete verifiable example] with:

- playbook
- output of running ansible-lint
- if you're getting a stack trace, output of
  `ansible-playbook --syntax-check playbook`

```paste below

```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by ajinkyau** [comment]
Hi @nrobert13 

`ansible-lint` works the same way as the `ansible` command.
Whenever you run it, it will look for related files in the current directory,
if it does not find it there, will look up the hierarchy in user or home directory,
and if it still doesn't locate these files it will fallback to the installation directory which has the defaults.

Per your information we see that it works correctly when you run it from the ansible directory which is the correct way to do it instead of the `repo` dir
